common/djangoapps/student/forms.py

```diff
@@ -144,6 +148,13 @@ class AccountCreationForm(forms.Form):
             "min_length": _PASSWORD_INVALID_MSG,
         }
     )
+    password_copy = forms.CharField(
```

Comment:
Instead of passing the copy of the password all the way to the back-end and storing it, why not just do validation on the front-end that the 2 fields have equal values (and disallow submission until they are equal)?